### PR TITLE
u3d/move command, renaming install dirs. Support long version names. Fixes #274

### LINKED ISSUE #274: Include build number when renaming hotfix releases of Unity
### Issue Checklist

- [X] I'm using the latest u3d version
- [X] I ran `u3d --help`
- [X] I read the [README](https://github.com/DragonBox/u3d/blob/master/README.md)
- [X] I made sure that a similar [issue](https://github.com/DragonBox/u3d/issues) doesn't exit

### Issue Description

We sometimes have hotfix releases installed in parallel to official releases. They'll have the same version, but they'll have a different revision code. Renaming them to the plain version number will not be possible because of the conflict to the official release.

When I install them manually I usually use the Unity_\<version>_\<revision> pattern.

Suggestion:
- check the installed revision to the official revision
- if they match, use the usual Unity_\<version> pattern
- if they don't match, use Unity_\<version>_\<revision>

The official revision seems to be used as part of the download url, i.e.:
> u3d available
> ...
> Version 5.4.1f1: https://download.unity3d.com/download_unity/**649f48bbbf0f**/
> ...

This revision is displayed in Unity in Help/About after pressing "Alt" on the keyboard.

At the start of the log file:
> Built from '5.4/release' branch; Version is '5.4.1f1 (**649f48bbbf0f**) revision 6594376'; Using compiler version '160040219'

Unfortunately I haven't found a good way to find this without running Unity, yet. On OSX it is in the UnityBuildNumber plist key. On Windows there seems to be a special "Unity Version" resource in the exe:

> BLOCK "StringFileInfo"
> {
> 	BLOCK "040904b0"
> 	{
> 		VALUE "CompanyName", "Unity Technologies ApS"
> 		VALUE "FileDescription", "Unity Editor"
> 		VALUE "FileVersion", "5.4.1.6594376"
> 		VALUE "LegalCopyright", "(c) 2016 Unity Technologies ApS. All rights reserved."
> 		VALUE "ProductName", "Unity"
> 		VALUE "ProductVersion", "5.4.1.6594376"
> 		VALUE "Unity Version", "5.4.1f1_**649f48bbbf0f**"
> 	}
> }
> 

This can be seen in "Resource Hacker". I am not a Ruby hacker, so I have no idea if there is a package to open .exe version resources...

### PR DESCRIPTION
**Pull request by lacostej**
Fixes #274 

I am splitting this PR in various pieces. This PR is based on top:

[MERGED]  #290 #291 #292 #293 #294 

#295 #296

### CODE REVIEW

File: lib/u3d_core/admin_tools.rb [hunk 1]
```diff
@@ -22,6 +22,17 @@
 ## --- END LICENSE BLOCK ---
 module U3dCore
   class AdminTools
+    def self.move_os_file(os, source_path, new_path, dry_run:)
+      if os == :windows
+        source_path.tr!('/', '\\')
+        new_path.tr!('/', '\\')
```
Comment (niezbop) [inline lib/u3d_core/admin_tools.rb]: Move `U3d::Utils.windows_path` to `U3dCore::Helper` and use it instead? It is a bit better than what you're using currently as it makes sure that specific paths will not cause issues (paths with numbers in IIRC).

**Comment by niezbop** [comment]
Most of it looks good to me! It's a bit hard to see the difference with all those PR right now, it will probably be easier to get a grasp of the modifications once this PR is rebased on the other.

Aside from the windows path translation, that I would like to see changed (let's not keep bug prone behaviours if we are aware they can break), I think that this is good to go.

As a side note, I believe that this PR is big enough to qualify for at least a minor version number change, and almost a major if we consider the sanitation part of the PR that is being restricted to list, although it could be seen as a bug.

**Comment by lacostej** [comment]
I rebased on top of master after merging all the other PRs. Gosh I love git.

I extracted the first 2 patches into #295 which you can review. I think we can add your change as a separate PR as it changes functionality, vs just refactoring now. WDYT?

**Comment by niezbop** [comment]
Git FTW :)

Agreed for the separate PR. I'm not on my computer so I can't do this now though

**Pull request merged**